### PR TITLE
Config the default blocksize to 32k for Python 3.7+

### LINKED ISSUE #11044: Single request uploads of large "readable" data stream are slow (capped at ~8Mbps)
- **Package Name**: 
azure-core (pipeline)
- **Package Version**: 
latest
- **Operating System**:
Windows 10 Enterprise (1909) . But this seems to be platform independent issue.
- **Python Version**: 
platform win32 -- Python 3.7.7, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- C:\git\azure-sdk-for-python\venv\Scripts\python.exe
- **References**
Root cause analysis brought me to these issues filled against Python httplib. Confirmed experimentally (see below).
https://bugs.python.org/issue21790
https://bugs.python.org/issue31945
https://stackoverflow.com/questions/48719893/why-is-the-block-size-for-python-httplibs-reads-hard-coded-as-8192-bytes

**Describe the bug**
When trying to push large amount of data (4000MB in my case) that's "readable stream" (e.g. BytesIO or file reader - anything implementing "read") then the upload speed caps at around 8Mbps.
(For the context I'm working on 4000MB block upload support for Azure Storage SDK).

**To Reproduce**
Execute [test_put_block_stream_large](https://github.com/Azure/azure-sdk-for-python/blob/ace5cc1cc18a38b12e5aacf4f46b4919cf970c36/sdk/storage/azure-storage-blob/tests/test_large_block_blob.py#L117-L134) with LARGE_BLOCK_SIZE bumped to some large value (i.e. 4000MB upcoming , or 100MB currently supported threshold).

OR

Use [scenario from my fork](https://github.com/kasobol-msft/azure-sdk-for-python/blob/149c9c67fb3ce8d240abfa0fa3da5a9a37253a34/sdk/storage/azure-storage-blob/tests/test_largest_block_blob.py#L105-L131) as reference.

**Expected behavior**
Upload speed of "readable" data is not capped by httpclient and can leverage full network bandwith available.

**Possible solution**
The https://bugs.python.org/msg305571 suggest quite handy workaround that could be part of pipeline I guess. So far I didn't see any way to inject different blocksize to httpclient.

**Screenshots**

Original test:

I was uploading 4000MB of data in single request without any modifications using "readable" stream. That took over 1 hour!!
image
image

Turns out httpclient is using 8192 byte buffer when readable stream is passed:

image
image
image

Then I started to play with blocksize. I was editing http client's source and bumping the blocksize.

After bumpting to 8192*1024 upload speed was more than 2X faster
image
image

And after bumping it to 10*8192*1024 I managed to upload that payload in ~4 and half minute.
image

**Additional context**
This is going to impact future users of "large block"/"large blob" (4000MB new limit for single block / 200TB limit for single blob). Users of that feature are most likely work with streams - either uploading data from network or data produced on the fly by computations. Therefore it's important to address this deficiency.

### PR DESCRIPTION
**Pull request by xiangyan99**
Fixes #11044

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/transport/_adapters.py [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+        password = password.encode('latin1')
+
+    authstr = 'Basic ' + to_native_string(
+        b64encode(b':'.join((username, password))).strip()
+    )
+
+    return authstr
+
+
+class HTTPAdapter(BaseAdapter):
```
Comment (hectorhdzg) [inline sdk/core/azure-core/azure/core/pipeline/transport/_adapters.py:146]: Is this adapter autogenerated code by swagger?

Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/pipeline/transport/_adapters.py:146]: No. This is a customized version of requests.adapters.HTTPAdapter. requests.adapters.HTTPAdapter does not accept kwargs in send method so we cannot specify block_size. In order to work around a performance issue in _socket, we want to be able to specify block_size when creating socket.

**Comment by mikeharder** [comment]
Perf results show Stream matches Array after this PR:

 &nbsp; |Stream (Mbps)| &nbsp; |Array (Mbps)| &nbsp; |
-----|-----|-----|-----|-----
**azure-core**|**Pipeline**|**stage\_block**|**Pipeline**|**stage\_block**
1.8.2|5.93|5.88|287|302
PR|271|302|270|284

Test environment: https://github.com/Azure/azure-sdk-for-python/issues/11044#issuecomment-690766623

**Review by kasobol-msft — DISMISSED** [review]

**Comment by xiangyan99** [comment]
I tried it on Linux and did not see any perf impact.

**Review by lmazuel — APPROVED** [review]

**Pull request merged**